README.md

```diff
@@ -1,5 +1,47 @@
+
+## Current Status
+Magda is currently being actively developed, and is still in an immature state - we've yet to fully stabilise and document our APIs or provide administration functionality outside of manually changing things via helm and kubectl. This should change very soon - currently we're developing an admin interface that will make it easy to administrate Magda without having to know Kubernetes, and we're working to make installation an extremely easy process once you already have a Kubernetes setup. If you want to jump in and give it a go feel free - we've been running some form of this in production for some time and haven't had any problems.
+
+## Architecture
+Magda is built around a collection of microservices that are distributed as docker containers. This was done to provide easy extensibility - Magda can be customised by simply adding new services using any technology as docker images, and integrating them with the rest of the system via stable HTTP APIs. Using Kubernetes for orchestration means that configuration of a customised Magda instance can be stored and tracked as plain text, and instances with identical configuration can be quickly and easily reproduced.
+
+![Magda Architecture Diagram](doc/magda-basic-architecture.png)
+
+### Registry
+Magda revolves around the _Registry_ - an unopinionated datastore built on top of Postgres. The Registry stores _records_ as a set of JSON documents called _aspects_. For instance, a dataset is represented as a record with a number of aspects - a basic one that records the name, description and so on as well as more esoteric ones that might not be present for every dataset, like temporal coverage or determined data quality. Likewise, distributions (the actual data files, or URLs linking to them) are also modelled as records, with their own sets of aspects covering both basic metadata once again, as well as more specific aspects like whether the URL to the file worked when last tested.
+
+Most importantly, aspects are able to be declared dynamically by other services by simply making a call with a name, description and JSON schema. This means that if you have a requirement to store extra information about a dataset or distribution you can easily do so by declaring your own aspect. Because the system isn't opinionated about what a record is beyond a set of aspects, you can also use this to add new entities to the system that link together - for instance, we've used this to store projects with a name and description that link to a number of datasets.
+
+### Crawlers
```

Comment:
Connectors? ;)

Reply:
ARGRGERHGGRHG

![panda flip](https://media.tenor.com/images/bb826e2c9c01bc28f9e00dc05ee054e3/tenor.gif)

I think I have some kind of psychological issue that prevents me from seeing the difference unless someone has already pointed it out